client/blocks/jitm/templates/modal-style.scss

```diff
@@ -55,18 +57,17 @@ $wpcom-modal-footer-height: 30px + ( $wpcom-modal-footer-padding-v * 2 );
 
 	.components-guide__container {
 		margin-top: 0;
+		min-height: unset;
 	}
 
 	.components-guide__footer {
 		display: none;
 	}
 
 	.components-guide__page {
-		position: absolute;
```

Comment:
The combination of the style definition of `.component-guide__page`, `.components-guide__container`, and `.modal__container` seemed to be the culprit for the broken style on mobile.